consoles/video_stream.pm

```diff
@@ -0,0 +1,340 @@
+    my $seconds_per_keypress = 1 / ($bmwqemu::vars{STREAM_TYPING_LIMIT} || STREAM_TYPING_LIMIT_DEFAULT);
+
+    # further slow down if being asked for.
+
+    # Note: the intended use of max_interval is the bootloader. The bootloader
+    # prompt drops characters when typing quickly. This problem mostly occurs
+    # in the bootloader. Humans notice because they look at the screen while
+    # typing. So this loop should be replaced by some true 'looking at the
+    # screen while typing', e.g. waiting for no more screen updates 'in the
+    # right area'.  For now, just waiting is good enough: The slow-down only
+    # affects the bootloader sequence.
+    if (($args->{max_interval} // consoles::console::DEFAULT_MAX_INTERVAL) < consoles::console::DEFAULT_MAX_INTERVAL) {
+        # according to 	  git grep "type_string.*, *[0-9]"  on
+        #   https://github.com/os-autoinst/os-autoinst-distri-opensuse,
+        # typical max_interval values are
+        #   4ish:  veeery slow
+        #   15ish: slow
+        $seconds_per_keypress = $seconds_per_keypress + 1 / sqrt($args->{max_interval});
+    }
+
+    for my $letter (split("", $args->{text})) {
+        next if ($letter eq "\r");
+        $self->_send_key_event($CHARMAP->{$letter} || $letter);
+        $self->{backend}->run_capture_loop($seconds_per_keypress);
+    }
+    return {};
+}
+
+sub send_key ($self, $args) {
+    $self->_send_key_event($args->{key});
+    $self->backend->run_capture_loop(.2);
+    return {};
+}
+
+sub _send_key_event ($self, $key) {
+    return unless $self->{input_pipe};
+    $self->{input_pipe}->write($key . "\n")
+      or die "failed to send '$key' input event";
+}
+
+sub get_last_mouse_set ($self, @) {
+    return $self->{mouse};
+}
+
+sub _mouse_move ($self, $x, $y) {
+    die "need parameter \$x and \$y" unless (defined $x and defined $y);
+    return unless $self->{input_pipe};
+
+    if ($self->{mouse}->{x} == $x && $self->{mouse}->{y} == $y) {
+        # in case the mouse is moved twice to the same position
+        # (e.g. in case of duplicated mouse_hide), we need to wiggle the
+        # mouse a bit to avoid qemu ignoring the repositioning
+        # because the SUT might have moved the mouse itself and we
+        # need to make sure the mouse is really where expected
+        my $delta = 5;
+        # move it to the left in case the mouse is right
+        #$delta = -5 if $x > $self->{width} / 2;
+        $delta = -5 if $x > 1024 / 2;
+        $self->_mouse_move($x + $delta, $y);
+    }
+
+    bmwqemu::diag "mouse_move to $x, $y";
+    $self->{input_pipe}->write("mouse_move $x $y\n");
+    $self->{input_pipe}->flush;
+
+    $self->{mouse}->{x} = $x;
+    $self->{mouse}->{y} = $y;
+    # let the event be processed before further commands
+    $self->backend->run_capture_loop(.1);
+
+    return;
+}
+
+sub mouse_hide ($self, $args) {
+    $args->{border_offset} //= 0;
+
+    #my $x = $self->{width} - 1;
+    #my $y = $self->{height} - 1;
```

Comment:
Don't add disabled code as nobody but you will understand why this is not active. Better leave it out and keep in your local git repository, either in `git stash` or a temporary "WIP"-commit

See https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/CONTRIBUTING.md#coding-style for more details